ci/jenkins_build

```diff
@@ -50,9 +67,164 @@ docker run --rm \
+    echo "INFO Checking if $kind were deleted"
+    resources=$(kubectl get $kind \
+      --namespace="$NAMESPACE" \
+      -o=json \
+      | jq -r '.items[] | "\(.kind)/\(.metadata.name)"')
+
+    res_count=$(echo $resources | wc -w)
+
+    if [[ "$res_count" -eq 0 ]]; then
+      break
+    else
+      # Ignore service account default
+      if [[ "$resources" = "ServiceAccount/default" ]]; then
+        break
+      else
+        echo "INFO Remaining: $res_count"
+        echo "INFO $resources"
+
+        elapsed_time=$(( $(date +%s) - $start_time ))
+        if [[ "$elapsed_time" -gt "$timeout" ]]; then
+          exit 1
+        fi
+
+        sleep "$poll_interval"
+      fi
+    fi
+  done
+}
+
+announce "Running build and tests..."
 pushd ../
   trap popd EXIT
-  ./build.sh
+
+  TEST_ID="$(random_string)"
+  export NAME="conjur"
+  export NAMESPACE="$NAME-$TEST_ID"
+  announce "Creating namespace $NAMESPACE"
+  kubectl create namespace "$NAMESPACE"
+  namespace_deleted=false
+  display_manifests=true
+
+  announce "Deploying Conjur application"
+  ./build.sh --persist
+
+  announce "Waiting for deployer to complete..."
+  deployer_name="${NAME}-deployer"
+  start_time=$(date +%s)
+  poll_interval=4
+  while true; do
+    deployer_status=$(kubectl get "jobs/$deployer_name" \
+                      --namespace="$NAMESPACE" -o=json | \
+                      jq '.status' || echo "{}")
+    failure=$(echo $deployer_status | jq '.failed // empty')
+    if [ -n "$failure" ]; then
+      clean_and_exit "Deployer failed"
+    fi
+
+    success=$(echo $deployer_status | jq '.succeeded // empty')
+    if [ -n "$success" ]; then
+      echo "INFO Deployer job succeeded"
+      break
+    fi
+
+    elapsed_time=$(( $(date +%s) - $start_time ))
+    printf "Elapsed ${elapsed_time}s\r"
+    if [[ "$elapsed_time" -gt "$WAIT_TIMEOUT" ]]; then
+      clean_and_exit "Deployer job timeout"
+    fi
+
+    sleep "$poll_interval"
+  done
+
+  announce "Checking test results"
+  pods=$(kubectl get pods \
+         --namespace="$NAMESPACE" \
+         --selector "app.kubernetes.io/name=conjur" \
+         --field-selector=status.phase=Failed \
+         --output=jsonpath={.items..metadata.name})
+  if [ -n "$pods" ]; then
+      clean_and_exit "Test failed. Found failed pods: [$pods]"
+  fi
+
+  announce "Tests passed, collecting logs"
+  # Set deployer name to empty after collecting logs so clean up doesn't try
+  # to collect logs again.
+  kubectl logs "jobs/$deployer_name" \
+      --namespace="$NAMESPACE" \
+      || (deployer_name=""; clean_and_exit "Failed to get logs for deployer $deployer_name")
+  deployer_name=""
+
+  announce "Delete the \"$NAME\" application"
+  kubectl delete "application/$NAME" \
+      --namespace="$NAMESPACE" \
+      || clean_and_exit "Failed to stop application"
+
+  announce "Wait for resources to be deleted from the namespace"
+  deletion_timeout=180
```

Comment:
If this is the only value we use, you can either make it a top-level constant or embed it into `wait_for_deletion` (I'm not suer if anything else uses this method).